client/src/components/Login/Register.js

```diff
@@ -57,6 +65,7 @@ const Register = () => {
   const handleSubmit = async (e) => {
     e.preventDefault();
     if (confirmPassword !== registerData.password) {
+      clearData(false);
       setErr("Password does not match");
       return;
     }
```

Comment:
Should we also call `cleardata` below if `!acceptTerms`?

Reply:
I though of calling the **`cleardata`** if `!acceptTerms`, but i think its just a warning that used needs to add select the terms and conditions, that why I have not added this. 

In this case suppose, user not selected the terms and conditions but still click on submit and we delete his password, that will be little strange experience to user. 

If you okay with this change, I will add the `clearData` if `!acceptTerms`.

Looking forward to your feedback. Thanks.

Reply:
That's a good point, in that case I'd say let's leave it how you have it. What do you think @lugenx?

Reply:
thank you @kunal8411 and @tsimian, for working on this. i am probably missing something, but i have a question. what if we just make two changes in this file:
just changing `setRegisterData` in `clearData` like this.
```javascript
  const clearData = () => {
    // setRegisterData({ name: "", email: "", password: "" }); //<---removed
    setRegisterData((prevState) => ({ ...prevState, password: "" })); ///<----added
    setConfirmPassword("");
    setAcceptTerms(false);
    setErr("");
  };
  ```
  
  And calling `clearData`, in `handleSubmit` without removing the other call:
  ```javascript
   if (confirmPassword !== registerData.password) {
      clearData(); // <---- added
      setErr("Password does not match");
      return;
    }
```

do you guys think it would be any different?
also, do you think clearing only password (not name and email) from the form can bring security risk? if answer is no, and these changes are not missing some cases, i would like to simplify it a bit. i will do some more research but please let me know.

Reply:
I dont think so there will be any security risk of removing only password field from the form.
I was just saying, we dont need to clear the password if user has not selected the `terms and conditions checkbox`, that case we can avoid, other than that I dont see any security risk of removing password on click of submit.

Reply:
Yeah, I think that would be more concise @lugenx. As for your other point, I agree with @kunal8411 and don't see it being a security risk.

Reply:
yes, what i read after i write this that, component will unmount right after the page routed, even clearing password is for any case. so i think we can simplify it.